optimum/exporters/openvino/__main__.py

```diff
@@ -444,12 +439,14 @@ class StoreAttr(object):
 
         from optimum.intel.openvino.quantization import _weight_only_quantization
 
+        submodel = core.read_model(submodel_path)
```

Comment:
if the model is gonna be read either ways, why nesting `return_model_sizes` and modify the behavior of all the export functions ? please explain the necessity for making export functions return model sizes as well, because it doesn't make sense for me.

Reply:
the porpose of this read model is prepare model for quantization, if model not planned to be compressed, there is no need in this step.
There are several cases for this:
1. nncf is not available
2. quantization disabled by export config (e.g. load_in_8bit=False in  from_pretrained or --weight-format fp16/fp32 in optimum-cli
3. default decision about necessary to compress model based on model size, we have rule by default apply int8 weight compression only in case if model size > 1B parameters and any additional quantization configs not provided (it is also the reason why we need calculate model size and avoid extra reading model if at this moment we already know that weight compression is not applicable)

Reply:
I also agree that avoiding an extra model read in case the model is not going to be quantized anyway is a good thing.

Reply:
Introducing `return_model_sizes` introduces some complexity so would prefer to avoid if possible, what do you think about moving this logic (applying quantization on each submodels depending on multiple criteria) from `main_export` to `_save_model` (or `export_pytorch` / `export_pytorch_via_onnx` / `export_tensorflow`) ? Also the `ov_config` could be updated to know whether we should check the model's size (in case where quantization wasn't specified) or if the model should or shouldn't be quantized (`load_in_8bit` set to `True` of `False` for example) wdyt @eaidova ?

Reply:
no, this is also thing what we wanted to avoid, when we save ov model, pytorch model still in memory and this memory can not be deallocated until we do not return into place where it was loaded (export_main) that lead to extra memory consumption